clients/python-wrapper/lakefs/branch.py

```diff
@@ -124,6 +124,24 @@ def get_commit(self):
         self._commit = None
         return super().get_commit()
 
+    def cherry_pick(self, reference: ReferenceType, parent_number: Optional[int] = None) -> Commit:
+        """
+        Cherry-pick a given reference onto the branch.
+
+        :param reference: ID of the reference to cherry-pick.
+        :param parent_number: When cherry-picking a merge commit,
+            the parent number (starting from 1) relative to which to perform the operation.
+            The destination branch is parent 1, which is the default behaviour.
+        :return: The cherry-picked commit at the head of the branch.
+        :raise NotFoundException: If either the repository or target reference do not exist.
+        :raise NotAuthorizedException: If the user is not authorized to perform this operation.
+        :raise ServerException: For any other errors.
+        """
+        ref = reference if isinstance(reference, str) else reference.id
+        cherry_pick_creation = lakefs_sdk.CherryPickCreation(ref=ref, parent_number=parent_number)
+        with api_exception_handler():
+            return self._client.sdk_client.branches_api.cherry_pick(self._repo_id, self._id, cherry_pick_creation)
```

Comment:
You need to wrap this with the lakeFS wrapper, see `uncommitted` example

Reply:
Done in [01d9ada](https://github.com/treeverse/lakeFS/commit/01d9adab5d594458fe1ef742c19f109a5233b3ea).